Implemented results tab and MCQ creation UI

shiquan-lim opened the pull request:
Completed function previously mentioned, of implementing the results tab and also restricting access to it based on whether the user is the event creator. 

Also implemented UI for creation of MCQ challenge. Challenge creation now has a switch to enable beta functionality.

shiquan-lim committed:
Implemented support for viewing results of text challenges. closes #36

shiquan-lim committed:
Merge branch 'master' of https://github.com/singpath/classmentors

shiquan-lim committed:
Made slight changes to textual result layout

atwj committed:
added other radio option buttons for challenges

todo:
——————
1. Find better way to manage inline css.
2. Add controllers for new option buttons

atwj committed:
Added behavior on MCQ radio button select

atwj committed:
Added todo

atwj committed:
Merge pull request #54 from shiquan-lim/issue-#31

Added ui for challenge creation, close #31

shiquan-lim committed:
Working on result display table

shiquan-lim committed:
Merge branch 'master' of https://github.com/shiquan-lim/classmentors

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into create_survey_form_#34

Commit:
Added survey checkbox and heading

Commit:
Updated gitignore to exclude idea files

sheryltohxn committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into issue-#23

Commit:
Created survey form html page for admin

Commit:
Made minimal changes to events.js

added a html file for routing for trying out, and delete it after a
failed attempt

atwj committed:
Added routing logic for various challenges

TODO:
—————————
1. Add routing paths to various challenges
2. Save state + details from previous forms prior to going into the
various challenges

atwj committed:
Added challenge.js

atwj committed:
Added mcq views, logic

atwj committed:
Merge branch 'master' into issue-#55

Commit:
created and deleted a html form

created and deleted a html form which I built for trying out purpose

atwj committed:
Added test code for mcq view

atwj committed:
Added routing for MCQ

atwj committed:
Added BETA functions button

shiquan-lim committed:
Merge pull request #56 from shiquan-lim/issue-#55

close #55

sheryltohxn committed:
gitignore

ignoring .iml & .xml files.

sheryltohxn committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into issue-#23

sheryltohxn committed:
added dialog box to cancel button #23

UX purposes; to ensure that user really wish to go back (and discard
data input)

shiquan-lim committed:
Merge pull request #57 from shiquan-lim/issue-#23

Added dialog box to cancel button   #23

atwj committed:
Merge remote-tracking branch 'singpath/master'

sheryltohxn committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into issue-#23

sheryltohxn committed:
Touch up on the dialog box

cosmetic changes

sheryltohxn committed:
Edits to comments

edited the comments in .js and .html

sheryltohxn committed:
Added view for MCQ

added draft look and feel for MCQ (without logic and validation)

sheryltohxn committed:
Disable users from adding option if single answer

atwj committed:
BETA functions enabled by default

atwj committed:
Easier to know which are BETA functions now

sheryltohxn committed:
Cosmetic Adjustments

sheryltohxn committed:
Added dropdown list

Commit:
Added routing to survey form

Created routing to survey form

atwj committed:
Added events service factory

*TODO*
- Fix breaking bug in service.

```
angular.js:13708 TypeError: eventCaptureService is not a function
    at new AddEventTaskCtrl
(https://localhost:8081/classmentors/components/events/events.js!transpi
led:549:5)
    at Object.instantiate
(https://localhost:8081/jspm_packages/github/angular/bower-angular@1.5.7
/angular.js:4724:14)
    at $controller
(https://localhost:8081/jspm_packages/github/angular/bower-angular@1.5.7
/angular.js:10249:28)
    at Object.link
(https://localhost:8081/jspm_packages/github/angular/bower-angular-route
@1.5.7/angular-route.js:1050:26)
    at eval
(https://localhost:8081/jspm_packages/github/angular/bower-angular@1.5.7
/angular.js:1240:18)
    at invokeLinkFn
(https://localhost:8081/jspm_packages/github/angular/bower-angular@1.5.7
/angular.js:9814:9)
    at nodeLinkFn
(https://localhost:8081/jspm_packages/github/angular/bower-angular@1.5.7
/angular.js:9215:11)
    at compositeLinkFn
(https://localhost:8081/jspm_packages/github/angular/bower-angular@1.5.7
/angular.js:8510:13)
    at publicLinkFn
(https://localhost:8081/jspm_packages/github/angular/bower-angular@1.5.7
/angular.js:8390:30)
    at lazyCompilation
(https://localhost:8081/jspm_packages/github/angular/bower-angular@1.5.7
/angular.js:8728:25) <div ng-view="" layout="column" flex=""
class="ng-scope" data-ng-animate="1">
```

shiquan-lim committed:
Merge pull request #58 from shiquan-lim/issue-#23

Drafted MCQ layout #23

sheryltohxn committed:
attempt to add input option field

not successful

sheryltohxn committed:
Revert "attempt to add input option field"

This reverts commit 9ebbc1ee08b3a01ab30036bf81b8d104a2072636.

sheryltohxn committed:
Revert "Revert "attempt to add input option field""

This reverts commit 5910ed3e285808fb9961644b27cb108e28027e9a.

sheryltohxn committed:
attempts to add option field dynamically

not successful

sheryltohxn committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into issue-#23

sheryltohxn committed:
Added ability to add new options and questions into mcq challenge

sheryltohxn committed:
Disable user from deleting last question

shiquan-lim committed:
Merge pull request #59 from shiquan-lim/issue-#23

#23 Refined MCQ creation page

shiquan-lim committed:
Restricted access to the result tab based on whether the user is the event creator. closes #48

dinoboff commented on src/classmentors/components/challenges/challenges.js:
```diff
@@ -0,0 +1,125 @@
+    //questions
+    var count_q_id = 1;
+
+    $scope.questions =[{
```
Why not `this.questions` and `$ctrl.questions` to access it?

shiquan-lim replied:
Noted. Please close this. We will send in another PR shortly

atwj committed:
Renamed eventCaptureService to eventService

atwj committed:
Added eventServiceFactory

atwj committed:
Testing out mcq.js

atwj committed:
Rendering out saved values from New Challenge View

atwj committed:
Created new challenge service factory

atwj committed:
Registered new service

atwj committed:
Debugging

atwj committed:
Added new save function to challengeServiceFactory

Commit:
#34 Survey form creation for admin view

#34 created survey form for admin view that can be added to firebase.

Commit:
Revert "#34 Survey form creation for admin view"

This reverts commit acc705d8cc789cba47372b8e3fc14d140f06ce50.

Commit:
Revert "Revert "#34 Survey form creation for admin view""

This reverts commit c267252dd618edbefa669a6b21634b7dfd22da5d.

Commit:
Revert "Revert "Revert "#34 Survey form creation for admin view"""

This reverts commit e81dfb4ffdcdea19cbdf0753b00d3334f30244d1.

Commit:
Revert "Revert "Revert "Revert "#34 Survey form creation for admin view""""

This reverts commit aab97e6aab07c5937e3d5aabeeb7ee60f5a0e41f.

Commit:
Revert "#34 Survey form creation for admin view"

This reverts commit acc705d8cc789cba47372b8e3fc14d140f06ce50.

Commit:
Revert "Added routing to survey form"

This reverts commit dccbd250c4cb27a8a53a65beee5fc2509e5cea3c.

Commit:
Merge remote-tracking branch 'origin/master'

atwj committed:
Completed controller logic for challenges.js

atwj committed:
event creation UI is one step or two step depending on task type

One step
——————————
1. Service / Badge / Problem
2. Link Pattern
3. Text Responses
4. Index Card
5. Code

Two-step
——————————
1. Multiple Choice
2. Video
3. Journalling

atwj committed:
Added two step form behavior to EditEventTaskCtrl

Known issues
——————————
1. Edit functionality for two step forms will end up creating a new
event. Will open it in another PR

atwj committed:
Merge branch 'master' into issue-#32, Resolve merge conflicts

# Conflicts:
#	src/classmentors/components/challenges/challenges.js
#	src/classmentors/components/challenges/mcq/mcq-view-mcq.html
#	src/classmentors/components/challenges/mcq/mcq.js
#	src/classmentors/components/events/events-view-event-task-form.html
#	src/classmentors/components/events/events.js

atwj committed:
Resolve merge conflicts

shiquan-lim committed:
Merge pull request #63 from shiquan-lim/issue-#32

close #32

shiquan-lim committed:
Enabled creation of code challenges. closes #61

shiquan-lim committed:
#61 bugfix for editing a code challenge. Previously when editing,the application memory will choose the task type of a code challenge as a text response challenge. Fixed.

shiquan-lim committed:
Added index cards as a beta function

shiquan-lim committed:
Cleaned up task form code

shiquan-lim committed:
Added control for code submission button. Working on #62

shiquan-lim committed:
Working on #62. Added new html template for handling code input.

shiquan-lim committed:
Working on #62. Edited size of dialog for code challenges.

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into create_survey_form_#34

# Conflicts:
#	src/classmentors/components/events/events-view-event-task-form.html
#	src/classmentors/components/events/events.js

Commit:
resolved conflicts from local branch

shiquan-lim committed:
Working on #62. Successfully integrated Ace editor into code challenges.

shiquan-lim committed:
#62, Added support for currently listed languages: Java, javascript, html, and python.

Commit:
#34 created survey form for admin

#34 created survey form for admin to upload and add into firebase

shiquan-lim committed:
Update .gitignore

shiquan-lim committed:
Merge pull request #64 from shiquan-lim/create_survey_form_#34

Create survey form for admin view close #34

shiquan-lim committed:
Working on #62. Implemented ng-model for code challenges.

shiquan-lim committed:
Pulled from master

shiquan-lim committed:
Revert "Working on #62. Implemented ng-model for code challenges."

This reverts commit 6e6c37339d176a60c269d6f4d1e1e1c7357b8275.

shiquan-lim committed:
Working on #62

shiquan-lim committed:
Revert "Working on #62. Implemented ng-model for code challenges."

This reverts commit 6e6c37339d176a60c269d6f4d1e1e1c7357b8275.

shiquan-lim committed:
Resolving conflict commit

shiquan-lim committed:
Merge pull request #65 from shiquan-lim/#61-#62-Code-Challenge

#61 #62 code challenge

shiquan-lim committed:
Working on #62. Code editing now functional

shiquan-lim committed:
closes #62. Also fixes bug of save buttom only working when enter is pressed. Now also works on click.

shiquan-lim committed:
Merge pull request #67 from shiquan-lim/#62-Refining-the-Code-Challenge

#62 refining the code challenge
